File: server/Infrastructure/LobTools/Models/AttachmentModels.cs
using System.Collections.Generic;

namespace Brainvest.Dscribe.LobTools.Models
{
	public class AttachmentsListRequest
	{
		public string EntityTypeName { get; set; }
		public int Identifier { get; set; }
	}

	public class AttachmentsListResponse
	{
		public IEnumerable<Item> Items { get; set; }

		public class Item
		{
			public string Description { get; set; }
			public int EntityTypeId { get; set; }
			public int Id { get; set; }
			public int Identifier { get; set; }
			public bool IsDeleted { get; set; }
			public string Title { get; set; }
			public string Url { get; set; }
			public string FileName { get; set; }
			public long Size { get; set; }
		}
	}

	public class DownloadAttachmentRequest
	{
		public int Id { get; set; }
	}
}

File: client/src/lib/lob-tools/models/attachment-models.ts
export class AttachmentListItem {
	Description: string;
	EntityTypeId: number;
	Id: number;
	Identifier: number;
	IsDeleted: boolean;
	Title: string;
	Url: string;
	FileName: string;
	Size: number;
}

export class AttachmentsListRequest {
	EntityTypeName: string;
	Identifier: number;
}

export class AttachmentsListResponse {
	Items: AttachmentListItem[];
}
